components/pool/RecentTrades.tsx

```diff
@@ -1,10 +1,112 @@
-import { memo } from 'react'
+import { memo, useState } from 'react'
+import NumberFormat from 'react-number-format'
+import { useQuery } from 'react-query'
 import styles from './styles/RecentTrades.module.scss'
 
+import { fetchPoolRecentSwaps } from 'lib/graphql'
+import { getSymbolFromAddress } from 'utils/address'
+import { timeAgo } from 'utils/date'
+import { truncateAddress } from 'utils/string'
+
+import { Jazzicon } from 'components/Jazzicon'
+import { Icon } from 'components/Icon'
+import { TokenIcon } from 'components/TokenIcon'
+
 function PoolRecentTrades() {
+  const [skip, setSkip] = useState(0)
+  const [swaps, setSwaps] = useState<Swap[]>([])
+
+  useQuery(['recentSwaps', skip], () => fetchPoolRecentSwaps(skip * 5, 5), {
```

Comment:
It seems like you're trying to implement a classic 'Load More' button, which can be written better with TanStack Query itself: https://tanstack.com/query/v4/docs/guides/infinite-queries

You can still filter out the duplicate items this way:
```typescript
Array.from(new Set(data.pages.flatten()))
```

Reply:
YESSSSS! `useInfiniteQuery` was what I was looking for. Thanks! 👍